test/ability_leaf_guard.c

```diff
@@ -0,0 +1,48 @@
+#include "global.h"
+#include "test_battle.h"
+
+SINGLE_BATTLE_TEST("Leaf Guard prevents non-volatile status conditions in sun")
+{
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_WILL_O_WISP].effect == EFFECT_WILL_O_WISP);
+        PLAYER(SPECIES_LEAFEON) { Ability(ABILITY_LEAF_GUARD); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_SUNNY_DAY); MOVE(opponent, MOVE_WILL_O_WISP); }
+    } SCENE {
+        ABILITY_POPUP(player, ABILITY_LEAF_GUARD);
+        NOT ANIMATION(ANIM_TYPE_MOVE, MOVE_WILL_O_WISP, opponent);
+        MESSAGE("It doesn't affect Leafeon…");
+        NOT STATUS_ICON(player, burn: TRUE);
+    }
+}
+
+SINGLE_BATTLE_TEST("Leaf Guard prevents non-volatile status conditions from items")
```

Comment:
```suggestion
SINGLE_BATTLE_TEST("Leaf Guard prevents burns from Flame Orb")
```

Reply:
If you wanna make this Flame Orb specific, should I make one for Toxic Orb too?

Reply:
More on this, if you want a test for Toxic Orb too, would it be sufficient to do this instead?

```c
SINGLE_BATTLE_TEST("Leaf Guard prevents status conditions from Flame Orb and Toxic Orb")
{
    u32 item;
    PARAMETRIZE { item = ITEM_FLAME_ORB; }
    PARAMETRIZE { item = ITEM_TOXIC_ORB; }
    GIVEN {
        ASSUME(gItems[ITEM_FLAME_ORB].holdEffect == HOLD_EFFECT_FLAME_ORB);
        ASSUME(gItems[ITEM_TOXIC_ORB].holdEffect == HOLD_EFFECT_TOXIC_ORB);
        PLAYER(SPECIES_LEAFEON) {Ability(ABILITY_LEAF_GUARD); Item(item); };
        OPPONENT(SPECIES_WOBBUFFET);
    } WHEN {
        TURN { MOVE(player, MOVE_SUNNY_DAY); }
    } SCENE {
        NONE_OF { 
            MESSAGE("Leafeon was burned!"); 
            MESSAGE("Leafeon is badly poisoned!");
            STATUS_ICON(player, burn: TRUE);
            STATUS_ICON(player, poison: TRUE);
        }
    }
}
```

Reply:
you can add an `if` to check for `item` value and change the test based on that.

Reply:
Like this? Or do I need to explicitly state `else if (item == ITEM_TOXIC_ORB)` as well?

```c
SINGLE_BATTLE_TEST("Leaf Guard prevents status conditions from Flame Orb and Toxic Orb")
{
    u32 item;
    PARAMETRIZE { item = ITEM_FLAME_ORB; }
    PARAMETRIZE { item = ITEM_TOXIC_ORB; }
    GIVEN {
        ASSUME(gItems[ITEM_FLAME_ORB].holdEffect == HOLD_EFFECT_FLAME_ORB);
        ASSUME(gItems[ITEM_TOXIC_ORB].holdEffect == HOLD_EFFECT_TOXIC_ORB);
        PLAYER(SPECIES_LEAFEON) {Ability(ABILITY_LEAF_GUARD); Item(item); };
        OPPONENT(SPECIES_WOBBUFFET);
    } WHEN {
        TURN { MOVE(player, MOVE_SUNNY_DAY); }
    } SCENE {
        if (item == ITEM_FLAME_ORB) {
            NONE_OF { MESSAGE("Leafeon was burned!"); STATUS_ICON(player, burn: TRUE);}
        }
        else {
            NONE_OF { MESSAGE("Leafeon is badly poisoned!"); STATUS_ICON(player, poison: TRUE);}
        }
    }
}
```

Reply:
Just an else should be enough